src/odb/include/odb/db.h

```diff
@@ -892,6 +892,18 @@ class dbBlock : public dbObject
     std::vector<dbBTerm*> bterms;
     bool order = false;
   };
+
+  struct dbBTermTopLayerGrid
+  {
+    dbTechLayer* layer = nullptr;
+    std::optional<int> x_step;
+    std::optional<int> y_step;
```

Comment:
What does it mean when the optional fields are not specified?  What is the `keepout`?  Please add some descriptive comments.

Reply:
I believe the `std::optional` change I've made yesterday doesn't really makes sense. All the files are mandatory when creating the top layer grid.

I'll add the comments for each field and remove the std::optional.